cmd/config-brancher/main.go

```diff
@@ -160,10 +160,40 @@ func removePeriodics(tests *[]api.TestStepConfiguration) {
 	}
 }
 
+func updatePromotion(currentConfig, futureConfig *api.ReleaseBuildConfiguration, currentRelease, futureRelease, devRelease string) {
+	if currentConfig.PromotionConfiguration == nil {
+		return
+	}
+
+	currentPromotion := currentConfig.PromotionConfiguration
+	futurePromotion := futureConfig.PromotionConfiguration
+
+	if currentPromotion.Name != "" || currentPromotion.Tag != "" {
+		// we cannot have two configs promoting to the same output, so
+		// we need to make sure the release branch config is disabled
+		futurePromotion.Disabled = futureRelease == devRelease
+	}
+
+	if currentPromotion.Targets == nil {
+		return
+	}
+
+	// filter and upgrade .promotion.to[] releases that promote to the current release
+	newTargets := make([]api.PromotionTarget, 0, len(currentPromotion.Targets))
+	for _, target := range currentPromotion.Targets {
+		if target.Name == currentRelease {
+			target.Name = futureRelease
+			target.Disabled = futureRelease == devRelease
+			newTargets = append(newTargets, target)
+		}
+	}
+	futurePromotion.Targets = newTargets
+}
+
 // updateRelease updates the release that is promoted to and that
 // which is used to source the release payload for testing
 func updateRelease(config *api.ReleaseBuildConfiguration, futureRelease string) {
-	if config.PromotionConfiguration != nil {
+	if config.PromotionConfiguration != nil && config.PromotionConfiguration.Name != "" {
```

Comment:
> You should check that the top-level fields (either name or tag) are set - if the determinize-ci-operator changes I originally suggested are live, no config in the repo should have them set, all should be using to - and, in that case, setting the top-level fields is not useful or wanted.

@stevekuznetsov  I guess it's the same here, right?

Reply:
Yes - and you'd need to check that both name *and* tag are `""`, since normal config only has to set one

Reply:
I don't quite get this one, as long as your changes to the `determinize-ci-operator` don't go live, either `.promotion.name` or `.promotion.tag` is set (exclude `.promotion.to[]` for now), doesn't it mean that I should set one of those two fields only if it's not empty?
```golang
currentPromotion := config.PromotionConfiguration
if currentPromotion != nil {
    if currentPromotion.Name != "" {
        currentPromotion.Name = futureRelease
    } else if currentPromotion.Tag != "" {
        currentPromotion.Tag = futureRelease
    }
}
```

Reply:
ACK, sorry - I misread your diff. I think the current PR is correct.